pkg/controller/direct/sql/sqlinstance_controller.go

```diff
@@ -317,7 +364,7 @@ func (a *sqlInstanceAdapter) Update(ctx context.Context, updateOp *directbase.Up
 	}
 
 	// Next, handle database edition updates
-	if !isEditionUnmanaged {
+	if !a.statefulFields["spec.settings.edition"].isUnmanaged {
```

Comment:
Might be better to keep the constants for this.

!a.statefulFields[unmanageEditionAnnotationValue].isUnmanaged.